### ISSUE TITLE
The default settings do not work correctly for nested key

### ISSUE THREAD

**Issue by altSoftLLC** [spf13/viper#209]
The default settings do not work correctly for nested key.
My patch (viper.go):

```golang
// Test for nested config parameter
    if strings.Contains(key, v.keyDelim) {
        path := strings.Split(key, v.keyDelim)
        source := v.find(path[0])
        if source != nil {
            if reflect.TypeOf(source).Kind() == reflect.Map {
                val := v.searchMap(cast.ToStringMap(source), path[1:])
                if val != nil {
                    jww.TRACE.Println(key, "found in nested config:", val)
                    return val
                }
            }
        }
    }
```

**Comment by lorin** [comment]
Here's a failing test case:

```
package main_test

import (
	"bytes"
	"testing"

	"github.com/spf13/viper"
)

func TestNestedDefault(t *testing.T) {
	data := `
	[main]
	enabled = false
	`

	viper.SetDefault("main.power", 100)
	viper.SetConfigType("toml")
	viper.ReadConfig(bytes.NewReader([]byte(data)))

	if got, want := viper.GetInt("main.power"), 100; got != want {
		t.Errorf(`got viper.GetInt("main.power")=%d, want %d`, got, want)
	}
}
```

**Comment by dh1tw** [comment]
I can confirm this bug. If a nested key is set via `setDefault(name, value)`, the value is stored as NIL. This can easily be confirmed by printing the settings: `viper.AllSettings()`

**Comment by GreysonD** [comment]
Any update on this?

Another example that fails:

```go
package main

import (
	"bytes"
	"github.com/spf13/viper"
)

func main() {
	jsonExample := []byte(`{
		"first": {
			"second": {
				"thing": "val"
			}
		}
	}`)

	viper.SetDefault("first.second.other", true)
	viper.SetConfigType("json")
	viper.ReadConfig(bytes.NewReader(jsonExample))

	println(viper.GetBool("first.second.other"))
	// Prints "false"
	// Should print "true"
}
```

**Comment by taybin** [comment]
Isn't this by design?  I believe you're setting not a nested type, but a single default value for the key "first.second.other".

**Comment by awishformore** [comment]
Fixed by #195
